Structure_Engine/Query/TorsionalConstant.cs

```diff
@@ -406,6 +497,118 @@ private static double AlphaLJunction(double tw, double tf, double r)
+
+            double alpha0 = AlphaTJunction(tw, t2, r);
+
+            //Equation 28
+            double alpha16 = -0.0836 + 0.2536 * tw / t2 + 0.1268 * r / t2 - 0.0806 * tw * r / Math.Pow(t2, 2) - 0.0858 * Math.Pow(tw / t2, 2); 
+            //Interpolate alpha between parallel flange and slope of 1 to 6:
+            return alpha0 + alpha16 * s / Math.Atan(0.1667);
+        }
+
+        /***************************************************/
+
+        [Description("Empirical formula used to correct the torsional constant with enhancement from a tapered L-junction such as a taper flange channel. Taken from Johnston & El Darwish, 1964")]
+        [Input("tw", "Web thickness.", typeof(Length))]
+        [Input("tf", "Mean flange thickness.", typeof(Length))]
+        [Input("r", "Root radius.", typeof(Length))]
+        [Input("t2", "Thickness of flange at intersection with near face of web.", typeof(Length))]
+        [Input("s", "Flange taper angle.", typeof(Length))]
+        private static double AlphaTaperLJunction(double tw, double tf, double r, double t2, double s)
+        {
+            if (0.2 > r / t2 || r / t2 > 1.0)
+                Reflection.Compute.RecordWarning("Calculation of alpha term of torsional constant may not be accurate because ratio of root radius to flange thickness is out of the applicable range.");
+
+            if (0.2 > tw / t2 || tw / t2 > 1.0)
+                Reflection.Compute.RecordWarning("Calculation of alpha term of torsional constant may not be accurate because ratio of web thickness to flange thickness is out of the applicable range.");
+
+            double alpha0 = AlphaLJunction(tw, tf, r);
+            //Equation 30
+            double alpha16 = -0.1325 + 0.3015 * tw / t2 + 0.1400 * r / t2 - 0.1070 * tw * r / Math.Pow(t2, 2) - 0.0956 * Math.Pow(tw / t2, 2); 
+            //Interpolate alpha between parallel flange and slope of 1 to 6:
+            return alpha0 + alpha16 * s / Math.Atan(0.1667);
+        }
+
+        /***************************************************/
+
+        private static double EndLossCorrectionVl(double s)
+        {
+            List<double> slopes = new List<double>
+            {
+                Math.Atan(0),
+                Math.Atan(1 / 50),
+                Math.Atan(1 / 20),
+                Math.Atan(1 / 6),
+            };
+
+            List<double> vl = new List<double>
+            {
+                 0.09045,
+                 0.10026,
+                 0.10307,
+                 0.10504,
+            };
+
+            return Interpolate(slopes, vl, s);
+        }
+
+        /***************************************************/
+
+        private static double EndLossCorrectionVs(double s)
+        {
+            List<double> slopes = new List<double>
+            {
+                Math.Atan(0),
+                Math.Atan(1 / 50),
+                Math.Atan(1 / 20),
+                Math.Atan(1 / 6),
```

Comment:
Also, forgot to add in the review, these all evaluate to 0 as it is assuming integer division 🤦 
Adding a `.0` at the end should help fix it.
```suggestion
                Math.Atan(0),
                Math.Atan(1 / 50.0),
                Math.Atan(1 / 20.0),
                Math.Atan(1 / 6.0),
```